lib/java/com/google/android/material/appbar/CollapsingToolbarLayout.java

```diff
@@ -261,6 +262,11 @@ public CollapsingToolbarLayout(@NonNull Context context, @Nullable AttributeSet
           a.getResourceId(R.styleable.CollapsingToolbarLayout_collapsedTitleTextAppearance, 0));
     }
 
+    // Now overlay any custom text Ellipsize
+    if (a.hasValue(R.styleable.CollapsingToolbarLayout_titleTextEllipsize)) {
+      setTitleEllipsize(a.getInt(R.styleable.CollapsingToolbarLayout_titleTextEllipsize, 2));
```

Comment:
Let's do something like setTitleEllipsize(convertEllipsizeToTruncateAt(a.getInt(R.styleable.CollapsingToolbarLayout_titleTextEllipsize, 2)) here?

Reply:
Also maybe make the (impossible) default value -1? So we can rely on the logic in the conversion method.

Reply:
Done, Added new method convertEllipsizeToTruncateAt() and set default value to -1